ocs_ci/framework/main.py

```diff
@@ -66,11 +70,66 @@ def init_ocsci_conf(arguments=None):
             os.path.expanduser(framework.config.RUN['bin_dir'])
         )
         utils.add_path_to_env_path(framework.config.RUN['bin_dir'])
+    print(framework.config)
     check_config_requirements()
 
 
+def sanitize_version_string(version):
+    """
+    Checks whether version string provided is of format x.y and returns
+    normalized string of form "x_y" which would serve as prefix for searching
+    version specific config files
+
+    Args:
+        version (str): input version by user (ex: 4.2, 4.3 etc)
+
+    Returns:
+        str: sanitized version string which would be prefix for default config
+            ex: 4_2, 4_3 etc
+
+    """
+    prefix = ''
+    for c in version.split('.'):
+        try:
+            int(c)
+        except ValueError:
+            logging.exception("Please provide proper version number")
+        prefix = f'{prefix}{c}_'
+    return prefix
+
+
+def init_version_defaults(arguments=None):
+    """
+    Update the config object with version specific defaults
+
+    Args:
+        arguments (list): Arguments for pytest execution
+
+    """
+    print(arguments)
+    parser = argparse.ArgumentParser(add_help=False, argument_default=argparse.SUPPRESS)
+    parser.add_argument('--ocs-version')
+    args, vals = parser.parse_known_args(arguments)
+    try:
+        if args.ocs_version:
+            ocs_version = args.ocs_version
+    except AttributeError:
+        ocs_version = "4.3"
```

Comment:
I think for now we still want 4.2 as default, maybe after GA

Reply:
done